### PR TITLE
f-enhancement/gh-198-login-token-expiration

### PR DESCRIPTION
**Pull request by aberonja**
Added token expiration handling.

GH issue: #198

### CODE REVIEW

File: src/main/frontend/src/store/sagas/HOC/withErrorHandling.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  return function* sagaWithErrorHandler(...args) {
+    try {
+      // yield* iterates over a generator and yields every value from it
+      // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/yield*
+      yield* saga(...args);
+    } catch (e) {
+      if (e.status === 401) {
+        yield put(push('/login'));
+      }
+      yield errorHandler();
```
Comment (iscekic) [inline src/main/frontend/src/store/sagas/HOC/withErrorHandling.js]:
```suggestion
      yield errorHandler(e);
```

File: src/main/frontend/src/store/sagas/HOC/withErrorHandling.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+          type: 'error',
+          title: 'Error',
+          message: e.message
+        })
+      );
+    }
+  };
+}
+
+export const errorHandlingAction = function*(action) {
```
Comment (iscekic) [inline src/main/frontend/src/store/sagas/HOC/withErrorHandling.js]:
```suggestion
export const genericErrorHandler = action => e => ...
```

Comment (iscekic) [reply src/main/frontend/src/store/sagas/HOC/withErrorHandling.js]: Usage afterwards:

```js
withErrorHandling(someSaga, genericErrorHandler(someAction));
```

File: src/main/frontend/src/store/sagas/getTokenSaga.js [hunk 3]
```diff
@@ -12,6 +13,9 @@ export const getToken = function*(action) {
     yield put(receiveToken(responseGoogle));
   } catch (error) {
     yield put(receiveTokenError(error));
+    if (error.status === 401) {
```
Comment (iscekic) [inline src/main/frontend/src/store/sagas/getTokenSaga.js]: Use `withErrorHandler` here as well

File: src/main/frontend/src/store/sagas/HOC/withErrorHandling.js (hunk 2 above)
Comment (aberonja) [reply src/main/frontend/src/store/sagas/HOC/withErrorHandling.js]: https://stackoverflow.com/questions/27661306/can-i-use-es6s-arrow-function-syntax-with-generators-arrow-notation

Comment (iscekic) [reply src/main/frontend/src/store/sagas/HOC/withErrorHandling.js]: You can still do:

```js
export const genericErrorHandler = action => function* (e) {}
```

**Pull request merged**